app-ios/tutanota/Sources/Contacts/ContactsSynchronization.swift

```diff
@@ -45,30 +63,35 @@ class ContactsSynchronization {
   func saveLocalContacts(_ contacts: [StructuredContact], forUserId userId: String) async throws {
     try await updateLocalContacts(contacts, forUserId: userId, deleteOldContacts: false)
   }
-  
+
   /// Remove a contact for the user.
   ///
   /// Parameters:
   /// - `contactId` defines the contact to remove.
   /// - `forUserId` denotes the user to modify.
-  func deleteLocalContact(_ contactId: String, forUserId userId: String) async throws {
+  func deleteLocalContact(_ contactId: String?, forUserId userId: String) async throws {
     guard await requestAuthorizationForContacts() else { return }
     let localContacts = getSavedLocalContactsIdentifierMappings(forUserId: userId)
-    guard let contactToDelete = localContacts[contactId] else { return }
-    try removeNativeContacts([contactId:contactToDelete], forUserId: userId)
+
+    if contactId != nil {
+      guard let contactToDelete = localContacts[contactId!] else { return }
+      try removeNativeContacts([contactId!: contactToDelete], forUserId: userId)
+    } else {
+      try removeNativeContacts(localContacts, forUserId: userId)
```

Comment:
i find this behavior very very surprising and not in good way. If it doesn't find a matching contact it just wipes all of them? is this what we want?

Reply:
It's not the case when it doesn't find a matching contacts, it's the case when no contactId is provided, meaning that we want to wipe out all contacts from given an user